users/models.py

```diff
@@ -1,3 +1,43 @@
+import os
+import time
+
+from PIL import Image
+from django.contrib.auth.models import User
 from django.db import models
 
-# Create your models here.
\ No newline at end of file
+from lqdoj_backend.settings import AVATAR_FOLDER
+
+
+def get_time_stamp():
+    # Get the seconds since epoch
+    seconds_since_epoch = time.time()
+    # Convert seconds since epoch to struct_time
+    time_obj = time.localtime(seconds_since_epoch)
+    return '%d-%d-%d_%d-%d-%d' % (
+        time_obj.tm_mday, time_obj.tm_mon, time_obj.tm_year, time_obj.tm_hour, time_obj.tm_min, time_obj.tm_sec)
+
+
+def get_file_path(instance, filename):
+    upload_to = os.path.join(AVATAR_FOLDER)
+    extension = filename.split('.')[-1]
+    filename = get_time_stamp() + "_" + instance.user.username
+    final_filename = '{}.{}'.format(filename, extension)
+
+    return os.path.join(upload_to, final_filename)
+
+
+class Profile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE, to_field="username")
+    avatar = models.ImageField(default='default.jpg', upload_to=get_file_path)
+
+    def __str__(self):
+        return "User profile: " + self.user.username
+
+    def save(self, force_insert=False, force_update=False, using=None,
+             update_fields=None):
```

Comment:
In fact, it's overriding. It's obligate to write an override function with the same args with the save function from super
I can replace it with `def save(self, **kwargs):` but didn't test. Anyway we have to pass the same args when override

Reply:
Probably something like this would work (references: https://stackoverflow.com/a/4269828/1851156)
```python
def save(self, *args, **kwargs):
    super().save(*args, **kwargs)
```

Reply:
Tested, it worked. 
In fact, this comment help me to realize a bug that will cause HTTP_500 when user try to update information. What a Quack debug!